src/metabase/plugins.clj

```diff
@@ -17,41 +18,49 @@
   (or (env/env :mb-plugins-dir)
       (.getAbsolutePath (io/file "plugins"))))
 
-;; logic for determining plugins dir -- see below
-(defonce ^:private plugins-dir*
-  (delay
-    (let [filename (plugins-dir-filename)]
-      (try
-        ;; attempt to create <current-dir>/plugins if it doesn't already exist. Check that the directory is readable.
-        (let [path (u.files/get-path filename)]
-          (u.files/create-dir-if-not-exists! path)
+(def ^:private plugins-dir*
+  (memoize/memo
```

Comment:
Changed this from a `defonce` to a function memoized on the result of `plugins-dir-filename` to facilitate testing